runtime/util/fltconv.c

```diff
@@ -386,8 +387,72 @@ helperCConvertLongToFloat(I_64 src)
 {
 	jfloat tmpDst;
 
+/* Every platform but 32-bit Windows can use the cast.
+ * Using VS2017 15.9 the cast doesn't give the expected result.
+ */
+#if !defined(WIN32) || defined(J9VM_ENV_DATA64)
 	tmpDst = (jfloat)src;
-
+#else /* !defined(WIN32) || defined(WIN64) */
```

Comment:
The comment here and on the `#endif` no longer match the `#if`.

Reply:
Updated.